### DISCUSSION TITLE
LoRAs, Embeddings, Perplexity evaluation and extensions, what, why, how?

### DISCUSSION THREAD

**Discussion by cibernicola** [oobabooga/text-generation-webui#2244] [Q&A]
Hello

I'm testing, like many around here :)
I have conceptual and half technical doubts, about LoRA and Embeedings in "oobabooga/text-generation-webui":

The conceptual ones are:

- Differences/definitions between LoRA and Embedding.
- Use cases for each.
- Any "assumable" explanation about Perplexity evaluation"?

The "half" technical ones:

 - Is there a possibility to generate/load embeddings in "oobabooga/text-generation-webui"?
    If no, why not?
    If yes, how?
 - I have tried to train LoRA's with the default configuration and no errors, but I have not seen big changes in the responses nor in the "way of acting":/

- Extensions:

When starting the webUI with the extension flag or enabling it in "interface mode", it does not imply that it is installed right? it is only activated, the installation must be done manually in the env, is it so? (if so I think it should be better informed about it).

I try to order concepts to understand them better, that's all :)

**Comment by maggenti** [comment]
Well, you are not the only one. I increased by four times the Rank, Alpha, but left the cutoff alone, and it took my file from 117kb to 970kb, but nothing changed in the model. It could not answer a single question put to it correctly. The Lora is broken or something. It is not loading correctly, for sure. And I updated it as of today, May 24th, 2023. I just do not get it... If it does not work, then please just grey out that part until you fix it or take it out. Thanks for reading this far.

**Reply by cibernicola** [reply]
Any response :(
